torchbenchmark/score/compute_score_v2.py

```diff
@@ -0,0 +1,210 @@
+        self._name = test_name
+        self._test_type, self._model, self._device, self._mode = _parse_test_name(self._name)
+        self._task = _get_model_task(self._model)
+        self._stable = test_item["stable"]
+        self._norm = test_item["norm"]
+    @property
+    def name(self) -> str:
+        return self._name
+    @property
+    def test_type(self) -> str:
+        return self._test_type
+    @property
+    def model(self) -> str:
+        return self._model
+    @property
+    def device(self) -> str:
+        return self._device
+    @property
+    def mode(self) -> str:
+        return self._mode
+    @property
+    def category(self) -> str:
+        return type(self._task).__name__
+    @property
+    def domain(self) -> str:
+        return self._task.name
+    @property
+    def norm(self) -> float:
+        return self._norm
+    @property
+    def stable(self) -> bool:
+        return self._stable
+
+class TorchBenchV2Suite:
+    def __init__(self, norm):
+        self._tests = []
+        self._tests_dict = {}
+        self._threshold = norm["stable_threshold"]
+        self._target = norm["target"]
+        for test in norm["tests"]:
+            test_item = TorchBenchV2Test(test, norm["tests"][test])
+            self._add_test(test_item)
+    @property
+    def target(self) -> float:
+        return self._target
+    @property
+    def all_stable_tests(self) -> List[TorchBenchV2Test]:
+        return list(filter(lambda x: x.stable, self._tests))
+    @property
+    def threshold(self) -> float:
+        return self._threshold
+    def _add_test(self, test: TorchBenchV2Test):
+        self._tests.append(test)
+        self._tests_dict[test.name] = test
+    def get_test_by_name(self, name) -> TorchBenchV2Test:
+        return self._tests_dict[name]
+
+class TorchBenchScoreV2:
+    # ref_data: the object read from reference YAML file or benchmark json file
+    def __init__(self, ref_data, _spec_file, _target):
+        if not ref_data:
+            with open(TORCHBENCH_V2_REF_DATA) as ref_file:
+                ref_data = yaml.full_load(ref_file)
+        # Build the suite
+        self.suite = TorchBenchV2Suite(self._setup_benchmark_norms(ref_data))
+
+    def _get_test_delta_weight(self, ref_norm, data_norm):
+        delta = (ref_norm - data_norm) / ref_norm
+        # Not a valid signal because it is below threshold
+        if abs(delta) <= self.suite.threshold:
+            return 0.0
+        return delta * 100
+
+    def _get_delta_score(self, data_norm):
+        "Compute V2 delta score"
+        delta = 0.0
+        for test in self.suite.all_stable_tests:
+            ref_norm = test.norm
+            data_test_norm = data_norm["tests"][test.name]["norm"]
+            delta_weight = self._get_test_delta_weight(ref_norm, data_test_norm)
+            if delta:
+                print(f"Test {test.name} with delta: {delta}, before: {ref_norm}, after: {data_test_norm}")
+            delta += delta_weight
```

Comment:
`delta` is the sum of deltas of all tests (excluding the tests that don't show delta below the threshold).